### PR TITLE
Validate format of timeZone before calling system LoadLocation method

### PR DESCRIPTION
**Pull request by liggitt**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Adds format validation of time zones ahead of calling the LoadLocation system-dependent method.

This insulates us from platform differences in handling of edge case values (xref https://github.com/kubernetes/kubernetes/pull/114508#pullrequestreview-1273796845)

A unit test ensuring all listed IANA time zones pass this validation is added.

Updates of CronJobs which do not change time zone values are already tolerated.

#### Which issue(s) this PR fixes:
Makes CronJob validation behavior more consistent across platforms.

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

cc @soltysh @claudiubelu

### CODE REVIEW

**Comment by claudiubelu** [comment]
It seems there are a few cases which are failing:

```
{Failed  === RUN   TestTimeZones
    validation_test.go:2789: America/Ciudad_Juarez failed: [<nil>: Invalid value: "America/Ciudad_Juarez": unknown time zone America/Ciudad_Juarez]
    validation_test.go:2789: Europe/Kyiv failed: [<nil>: Invalid value: "Europe/Kyiv": unknown time zone Europe/Kyiv]
--- FAIL: TestTimeZones (0.18s)
```

File: pkg/apis/batch/validation/validation.go [hunk 1]
```diff
@@ -513,6 +525,13 @@ func validateTimeZone(timeZone *string, fldPath *field.Path) field.ErrorList {
 		return allErrs
 	}
 
+	for _, part := range strings.Split(*timeZone, "/") {
+		if part == "." || part == ".." || strings.HasPrefix(part, "-") || !validTimeZoneCharacters.MatchString(part) {
+			allErrs = append(allErrs, field.Invalid(fldPath, timeZone, fmt.Sprintf("unknown time zone %s", *timeZone)))
```
Comment (pacoxu) [inline pkg/apis/batch/validation/validation.go:530]: Here we already split it into parts, should we mention which part is an unknown timezone in the error message?

**Comment by pacoxu** [comment]
> validation_test.go:2789: America/Ciudad_Juarez failed: [<nil>: Invalid value: "America/Ciudad_Juarez": unknown time zone America/Ciudad_Juarez]
>     validation_test.go:2789: Europe/Kyiv failed: [<nil>: Invalid value: "Europe/Kyiv": unknown time zone Europe/Kyiv]

The test failed on Linux and passed on Mac.

**Comment by pacoxu** [comment]
```
root@16a10d869e1f:/usr/src/kubernetes# cat /usr/local/go/src/time/zoneinfo_test.go | grep TestBadLocationErrMsg -A 10
func TestBadLocationErrMsg(t *testing.T) {
        time.ResetZoneinfoForTesting()
        loc := "Europe/Kyiv"
        want := errors.New("unknown time zone " + loc)
        _, err := time.LoadLocation(loc)
        if err.Error() == want.Error() {
                t.Errorf("LoadLocation(%q) error = %v; want %v", loc, err, want)
        }
}

func TestLoadLocationValidatesNames(t *testing.T) {
root@16a10d869e1f:/usr/src/kubernetes# /usr/local/go/bin/go test -timeout 30s -run ^TestBadLocationErrMsg$ time
--- FAIL: TestBadLocationErrMsg (0.00s)
    zoneinfo_test.go:46: LoadLocation("Europe/Kyiv") error = unknown time zone Europe/Kyiv; want unknown time zone Europe/Kyiv
FAIL
FAIL    time    0.003s
FAIL
```

The reason seems to be that `America/Ciudad_Juarez` and `Europe/Kyiv` are valid on Mac and not Valid in linux.

**Comment by pacoxu** [comment]
**America/Ciudad_Juarez** 
http://mm.icann.org/pipermail/tz-announce/2022-November/000076.html
-  A new Zone America/Ciudad_Juarez splits from America/Ojinaga. It was added [recently](https://stackoverflow.com/questions/74810441/timezone-obtained-via-timezonefinder-america-ciudad-juarez-generates-error-u).

**Europe/Kyiv**
The suggestion to rename the time zone ID Europe/Kiev to Europe/Kyiv has been made and accepted! The change was introduced in release 2022b (Released 2022-08-11).
It can be found in https://data.iana.org/time-zones/tzdb-2022b/backward.

File: pkg/apis/batch/validation/validation.go [hunk 2]
```diff
@@ -502,6 +504,16 @@ func validateScheduleFormat(schedule string, timeZone *string, fldPath *field.Pa
...
 }
 
+// https://data.iana.org/time-zones/theory.html#naming
+// * A name must not be empty, or contain '//', or start or end with '/'.
+// * Do not use the file name components '.' and '..'.
+// * Within a file name component, use only ASCII letters, '.', '-' and '_'.
+// * Do not use digits, as that might create an ambiguity with POSIX TZ strings.
+// * A file name component must not exceed 14 characters or start with '-'
+//
+// 0-9 and + characters are tolerated to accomodate legacy compatibility names
```
Comment (pacoxu) [inline pkg/apis/batch/validation/validation.go:514]: `accommodate`

**Comment by liggitt** [comment]
> The reason seems to be that `America/Ciudad_Juarez` and `Europe/Kyiv` are valid on Mac and not Valid in linux.

those were added after go1.19 was released, pruned the unit test to just valid timezones as of go1.19 release, and imported "time/tzdata" to the unit test to mitigate cross-platform differences

File: pkg/apis/batch/validation/validation.go (hunk 1 above)
Comment (liggitt) [reply pkg/apis/batch/validation/validation.go:530]: I don't think there's much value in that granularity, they should check the overall value for validity

**Comment by liggitt** [comment]
Unit tests passed on Mac, windows, and Linux

**Pull request merged**